README.md

```diff
@@ -1,170 +1,189 @@
-                       │ (Event Creation) │
-                       └──────────────────┘
-```
+## Table of Contents
 
-The service processes incoming traces and broadcasts structured events to connected WebSocket clients.
+- [How It Works](#how-it-works)
+- [Prerequisites](#prerequisites)
+- [Getting Started](#getting-started)
+- [Configuration](#configuration)
+- [End-to-End (E2E) Testing](#end-to-end-e2e-testing)
+- [Testing Tools and Their Configuration](#testing-tools-and-their-configuration)
+- [API Reference and Integration](#api-reference-and-integration)
+- [Contributing](#contributing)
 
-## Implementation Details
 
-### Core Components
+-----
 
-#### FastAPI Application (`main.py`)
-Entry point that sets up the FastAPI app and includes routers. The actual endpoints are organized into separate route modules.
+## How It Works
 
-#### API Routes (`api/routes/`)
-- **`traces.py`**: OTLP trace ingestion endpoint that receives protobuf data and processes it into events
-- **`websockets.py`**: WebSocket endpoints for global and conversation-specific event streaming
+The service acts as a pipeline, processing telemetry data and making it available for real-time consumption.
 
-#### Core Business Logic (`core/`)
-- **`span_preprocessor.py`**: Converts OpenTelemetry spans into typed events by extracting semantic attributes, mapping span kinds to event types, and preserving tracing context
-- **`connection_manager.py`**: Manages WebSocket connections with global broadcast and conversation-specific routing
-- **`state.py`**: Global application state management for connection managers and shared resources
+1.  **Agent Instruments**: Your agentic application is instrumented with OpenTelemetry to emit traces.
+2.  **Ingestion Endpoint**: The service receives these traces at its OTLP/HTTP endpoint (`/v1/traces`).
+3.  **Span Processing**: A core preprocessor (`span_preprocessor.py`) parses the trace spans, extracting key attributes and mapping them to structured event models (e.g., `AgentStart`, `LLMCall`, `ToolOutput`).
+4.  **WebSocket Broadcast**: The processed events are immediately broadcasted through a connection manager to all subscribed WebSocket clients.
 
-#### Data Models (`models/`)
-- **`events.py`**: Structured dataclasses for communication events (agent lifecycle, LLM calls, tool invocations)
-- **`content.py`**: Content type definitions for LLM requests/responses and tool interactions
-
-#### Utilities (`utils/`)
-- **`extractors.py`**: Attribute extraction logic for parsing OTLP span data
-- **`factories.py`**: Factory functions for creating specific event types from span data
+-----
 
 ## Prerequisites
 
-- **Python 3.13+**: Required for the app
-- **uv**: Package manager for Python projects
-- **Kubernetes cluster**: Required for Tilt development (see Local Kubernetes Setup below)
+To build, run, and contribute to this project, you'll need the following tools installed.
 
-## Setup
+| Tool | Version | Notes |
```

Comment:
I'd remove the version. For Python, this is defined in pyproject.toml, docker is not really important.